test/controllers/users_controller_test.rb

```diff
@@ -135,6 +135,23 @@ class UsersControllerTest < ActionController::TestCase
     assert_redirected_to users_url
   end
 
+  test "current user have to enter current password to change password" do
+    user = User.new :login => "John", :mail => "john@doe.com", :firstname => "john", :lastname => "doe", :auth_source => auth_sources(:internal)
+    User.current = user
+    user.password = "password"
+    assert user.save
```

Comment:
Prefer `FactoryGirl.create(:user, :password => 'password')` over manual construction. (Actually, 'password' is even the default for the factory.)